migration-engine/core/src/lib.rs

```diff
@@ -115,10 +116,25 @@ pub async fn qe_setup(prisma_schema: &str) -> CoreResult<()> {
             ]
             .contains(&provider.as_str()) =>
         {
-            SqlMigrationConnector::qe_setup(&source.url().value).await?;
+            SqlMigrationConnector::new(&source.url().value).await?
         }
         x => unimplemented!("Connector {} is not supported yet", x),
-    }
+    };
+    let engine = MigrationEngine::new(connector).await?;
+
+    // reset
+    ResetCommand::execute(&(), &engine).await?;
+
+    // creates schema & database
+    SqlMigrationConnector::qe_setup(&source.url().value).await?;
+
+    // create the database schema for given Prisma schema
+    let schema_push_input = SchemaPushInput {
+        schema: prisma_schema.to_string(),
+        assume_empty: true,
+        force: true,
+    };
+    SchemaPushCommand::execute(&schema_push_input, &engine).await?;
```

Comment:
I think this can easily optimized a bit further by an ME expert. There still unnecessary steps in here imo.

Reply:
Yes, I think `SqlMigrationConnector::qe_setup()` could act as a constructor for `SqlMigrationConnector` in addition to the its own setup. It would then also include the reset, and the only step that would be left would be creating the engine and calling SchemaPush on it. But it's also something we can do later, this PR already looks like a huge win.